dom/events/passive-default/resources/default-passive-scrolling.js

```diff
@@ -0,0 +1,40 @@
+function waitFor(condition) {
+  const MAX_FRAME = 200;
+  return new Promise((resolve, reject) => {
+    function tick(frames) {
+      // We requestAnimationFrame either for 200 frames or until condition is
+      // met.
+      if (frames >= MAX_FRAME)
+        reject("Condition did not become true after 200 frames");
+      else if (condition())
+        resolve();
+      else
+        requestAnimationFrame(tick.bind(this, frames + 1));
+    }
+    tick(0);
+  });
+}
+
+function runTest({target, eventName, expectCancelable}) {
+  let cancelable = null;
+  let arrived = false;
+  target.addEventListener(eventName, function (event) {
+    cancelable = event.cancelable;
+    arrived = true;
+    event.preventDefault();
+  });
+
+  promise_test (async (t) => {
+    t.add_cleanup(() => {
+      document.querySelector('.remove-on-cleanup')?.remove();
+    })
+    const pos_x = Math.floor(window.innerWidth / 2);
+    const pos_y = Math.floor(window.innerHeight / 2);
+    const delta_x = 0;
+    const delta_y = 100;
+    await new test_driver.Actions()
+      .scroll(pos_x, pos_y, delta_x, delta_y).send();
+    await waitFor(()=> { return arrived; });
+    assert_equals(cancelable, expectCancelable);
```

Comment:
I think it's also worth testing `event.defaultPrevented`. From what I understand, an event might be `cancelable`, but because of the proposal we're testing here, its listeners will be passive, so even after calling `event.preventDefault()`, `event.defaultPrevented` will be false.

Right now it seems like an implementation could pass the tests by never doing anything regarding passive-ness, only cancelable-ness.

Reply:
You're right, I had misunderstood now passive works and thought it wasn't directly testable, but indeed it is testable with `defaultPrevented` and firing a cancelable event with `dispatchEvent`, like this:

 http://software.hixie.ch/utilities/js/live-dom-viewer/saved/10420